### PR TITLE
[IMP] web: convert mobile list_view test to hoot

### PR DESCRIPTION
**Pull request by res-odoo**
task-4028335

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/tests/views/list/list_view.test.js [hunk 1]
```diff
@@ -194,22 +199,52 @@ class Currency extends models.Model {
 
 defineModels([Foo, Bar, Currency, ResCompany, ResPartner, ResUsers]);
 
+async function clickRecordSelector() {
+    if (getMockEnv().isSmall) {
+        await contains("tbody td.o_data_cell:eq(0)").drag();
+    } else {
+        await contains(`tbody tr.o_data_row[data-id] input`).click();
```
Comment (aab-odoo) [inline addons/web/static/tests/views/list/list_view.test.js:206]: `[data-id]` not necessary

Comment (aab-odoo) [reply addons/web/static/tests/views/list/list_view.test.js:206]: I would use `tbody .o_data_row .o_list_record_selector input` to be more explicit

File: addons/web/static/tests/views/list/list_view.test.js [hunk 2]
```diff
@@ -194,22 +199,52 @@ class Currency extends models.Model {
 
 defineModels([Foo, Bar, Currency, ResCompany, ResPartner, ResUsers]);
 
+async function clickRecordSelector() {
+    if (getMockEnv().isSmall) {
+        await contains("tbody td.o_data_cell:eq(0)").drag();
```
Comment (aab-odoo) [inline addons/web/static/tests/views/list/list_view.test.js:204]: drag? Also, why `eq(0)`here but not below?

File: addons/web/static/tests/views/list/list_view.test.js [hunk 3]
```diff
@@ -194,22 +199,52 @@ class Currency extends models.Model {
...
+    } else {
+        await contains(`tbody tr.o_data_row[data-id] input`).click();
+    }
+}
+
+async function selectAllRecord() {
```
Comment (aab-odoo) [inline addons/web/static/tests/views/list/list_view.test.js:210]: selectAllRecord**s**

File: addons/web/static/tests/views/list/list_view.test.js [hunk 4]
```diff
@@ -194,22 +199,52 @@ class Currency extends models.Model {
...
+    if (getMockEnv().isSmall) {
+        const cells = queryAll("tbody tr.o_data_row[data-id]:not(.o_data_row_selected)");
+        for (const cell of cells) {
+            await contains(cell).drag();
+        }
+    } else {
+        await contains(`thead .o_list_record_selector input`).click();
+    }
+}
+async function unselectAllRecord() {
```
Comment (aab-odoo) [inline addons/web/static/tests/views/list/list_view.test.js:220]: unselectAllRecord**s**

File: addons/web/static/tests/_framework/search_test_helpers.js [hunk 5]
```diff
@@ -286,7 +286,13 @@ export async function validateSearch() {
  * @param {import("./mock_server/mock_server").ViewType} viewType
  */
 export async function switchView(viewType) {
-    await contains(`button.o_switch_view.o_${viewType}`).click();
+    if (getMockEnv().isSmall) {
+        await contains(".o_cp_switch_buttons .dropdown-toggle").click();
+        //await animationFrame(); //needed?
```
Comment (aab-odoo) [inline addons/web/static/tests/_framework/search_test_helpers.js:291]: check with multi build and remove

**Review by adr-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**